install.sh

```diff
@@ -197,7 +197,7 @@ legacy_transform_after_update() {
     COMMIT_COUNT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-list --count HEAD)
     GITHOOKS_CLONE_CURRENT_COMMIT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-parse HEAD)
 
-    if [ "$COMMIT_COUNT" != "1" ] && [ -z "$INTERNAL_UPDATED_FROM_COMMIT" ]; then
+    if [ "$COMMIT_COUNT" != "1" ] && [ -z "${INTERNAL_UPDATED_FROM_COMMIT+set}" ]; then
```

Comment:
Can you explain what does this change?
Also, does this work in all Shell variants?

Reply:
This variable expansion is Posix I think.
INTERNAL_UPDATED_FROM_COMMIT is set to either "" (no update), "commitsha..." (update) or "0000..0000" (clone).
What we need to check here is: Is this variable *not* set, meanining we are before any changes where we introduces this stuf...

Reply:
> ${parameter:+word}
Use Alternative Value. If parameter is unset or null, null shall be substituted; otherwise, the expansion of word shall be substituted.

https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html

Reply:
Oh crap, its wrong lol

Reply:
Fixed.

Reply:
For me, this always makes the `-z` test false...

Reply:
Yes, this only gives me true if I unset a variable, but doesn't match if it's set to empty string.
I still don't get why is this a problem here, but I trust you here with this code better than my understanding of it.